### PR TITLE
Updated solidity smart contract to handle when the party is full

### PR DESCRIPTION
**Pull request by Mas9311**
Added a modifier, partyNotFullModifier which calls the function below
Added a function, partyFull which returns true if currentNumberOfGuests > maxGuestsPermitted.

When the last guest to be added is added, add owner to the list of consenters and finalizes the party.

Anytime before the last guest consents, the owner can still close the party (and not rendering their consent.
 - If owner decides to cancel, they must give a reason.
    - i.e.: "A guest I do not want to consent with was added to the party."

### CODE REVIEW

**Comment by Mas9311** [comment]
There, everything should be up to speed in step 3: joining Alice's party.
 1. Now checks to make sure the party has not expired yet (Party.closeTime)
 1. When guests permitted == current number of guests in party:
     - Add owner's consent to party, state is Finalized, and closeTime is now.
 1. Added a mandatory reasonForCancel if owner decides to cancel (must be done before the party is full).

Only _possible_ bug left in solidity is an implementation one: Can the owner cancel the party even the the time has expired (and guests are not full). Here's an example: Let's say A plans on a party of 2 guests: B and C. Only B joins, C never joins. 2 years later, the party is still not "finalized", per se. A _can_ still cancel the party, BUT the closeTime would be updated to the newest time (2 years after).
_Thoughts?_

**tl;dr**: only when the last person joins does the party get "finalized".
How do we handle the situation of: last person (in a 2+ party) never joins.
